src/main/java/at/hannibal2/skyhanni/features/event/carnival/CarnivalReminder.kt

```diff
@@ -0,0 +1,95 @@
+import at.hannibal2.skyhanni.events.SecondPassedEvent
+import at.hannibal2.skyhanni.skyhannimodule.SkyHanniModule
+import at.hannibal2.skyhanni.utils.ChatUtils
+import at.hannibal2.skyhanni.utils.HypixelCommands
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.RegexUtils.matches
+import at.hannibal2.skyhanni.utils.SimpleTimeMark
+import at.hannibal2.skyhanni.utils.SimpleTimeMark.Companion.fromNow
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import java.time.ZoneOffset
+import java.time.ZonedDateTime
+import kotlin.time.Duration.Companion.minutes
+import kotlin.time.Duration.Companion.seconds
+
+@SkyHanniModule
+object CarnivalReminder {
+
+    private val config get() = SkyHanniMod.feature.event.carnival
+    private val storage get() = ProfileStorageData.profileSpecific?.carnival
+
+    private var nextCheckTime = SimpleTimeMark.farFuture()
+
+    private var claimedToday = false
+
+    private var lastClaimedDay
+        get() = storage?.lastClaimedDay
+        set(value) {
+            storage?.lastClaimedDay = value
+        }
+
+    private val repoGroup = RepoPattern.group("carnival.tickets")
+
+    /** REGEX-TEST: §aYou claimed §r§aCarnival Ticket §r§8x25§r§a!
+     */
+    private val ticketClaimedPattern by repoGroup.pattern("claimed", "§aYou claimed §r§aCarnival Ticket §r§8x25§r§a!")
+
+    /** REGEX-TEST: §e[NPC] §aCarnival Leader§f: §rYou've already claimed your §aCarnival Tickets §ffor §btoday§f, but I'm happy to answer any questions you might have.
+     */
+    private val alreadyClaimedPattern by repoGroup.pattern(
+        "already",
+        "§e[NPC] §aCarnival Leader§f: §rYou've already claimed your §aCarnival Tickets §ffor §btoday§f, but I'm happy to answer any questions you might have.",
+    )
+
+    @SubscribeEvent
+    fun onSecondPassedEvent(event: SecondPassedEvent) {
+        if (!isEnabled() || nextCheckTime.isInFuture()) return
+        check()
+    }
+
+    @SubscribeEvent
+    fun onProfileJoin(event: ProfileJoinEvent) {
+        claimedToday = false
+        if (!isEnabled()) return
+        nextCheckTime = 30.0.seconds.fromNow()
+        check()
+    }
+
+    @SubscribeEvent
+    fun onLorenzChat(event: LorenzChatEvent) {
+        if (!isEnabled() && !claimedToday) return
+        if (!ticketClaimedPattern.matches(event.message) && !alreadyClaimedPattern.matches(event.message)) return
+        claimedToday = true
+        lastClaimedDay = ZonedDateTime.now(ZoneOffset.UTC).toLocalDate()
+    }
+
+    fun check() {
+        if (claimedToday) {
+            val currentDay = ZonedDateTime.now(ZoneOffset.UTC).toLocalDate()
+            val lastClaimedDay = lastClaimedDay
+
+            if (lastClaimedDay == null || currentDay.isAfter(lastClaimedDay)) {
+                claimedToday = false
+            }
+        } else {
+            ChatUtils.clickableChat(
```

Comment:
Please use `ChatUtils.clickToActionOrDisable()` instead here.

Reply:
Also maybe even check against `ReminderUtils.isBusy()`.

Reply:
Changed. And worked. (5min is just quiet long)
![image](https://github.com/user-attachments/assets/a5102954-22ff-475e-a084-d28edd40ddf0)